api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -35,4 +37,9 @@ message UdpProxyConfig {
   // The idle timeout for sessions. Idle is defined as no datagrams between received or sent by
   // the session. The default if not specified is 1 minute.
   google.protobuf.Duration idle_timeout = 3;
+
+  // Optional configuration for UDP proxy routing policy. It only supports :ref:`hash_policy<envoy_v3_api_msg_type.v3.HashPolicy>` routing policy.
+  // If route_policy is not set and intend to use hash-based load balancing algorithms,
+  // the hash-based load balancing algorithms will select a host randomly.
+  type.v3.UdpRoutePolicy route_policy = 4;
```

Comment:
Should this be repeated? This would allow you to have multiple policies that contribute to the hash in the future? Also should this be specific to hashing? `UdpHashPolicy` and `hash_policies`?

Reply:
Could you let me know more in detail? It is not repeated and not be specific to hashing. So, its name is route_policy because it can have one of hashing policy or some thing like as byte_range_policy.
Please let me know if I have wrong understanding. :)

Reply:
I think it is actually specific to hashing? What other type of route policies do you imagine having? I think the reason why we want repeated somewhere is you might want to hash on a byte range + IP address, for example.

Reply:
Sorry, I little confused that the byte range is another type of route policy because the proposal explained that the option will have like as 'x7zs9: true'.
So, I thought that if it is implemented, it will use memory comparison not hash comparison. :(

I have updated the api and implementation what you and me might want. :)
It wrapped the SourceIP option with the HashPolicy and it can be repeated, so it can be combined with later new policy.

Please check it once again. :)